simulation/halsim_ws_core/src/main/native/cpp/WSHalProviders.cpp

```diff
@@ -0,0 +1,57 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2017-2020 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#include "WSHalProviders.h"
+
+namespace wpilibws {
+
+void HALSimWSHalProvider::OnNetworkConnected(
+    std::shared_ptr<HALSimBaseWebSocketConnection> ws) {
+  {
+    std::lock_guard lock(mutex);
+
+    // store a weak reference to the websocket object
+    m_ws = ws;
+  }
+
+  // trigger a send of the current state
+  // -> even if this gets called before, it's ok, because we cleared the
```

Comment:
This comment is no longer needed... but heed it when sending all values to the client!